tcms/core/tests/test_forms.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name
+from django.test import TestCase
+from django.core.exceptions import ValidationError
+
+from tcms.core.forms.fields import UserField
+from tcms.tests.factories import UserFactory
+
+
+class TestUserField(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = UserFactory(username='admin', email='admin@example.com')
+        cls.user.set_password('admin')  # nosec:B106:hardcoded_password_funcarg
+        cls.user.is_superuser = True
+        cls.user.is_staff = True  # enables access to admin panel
+        cls.user.save()
+
+        cls.user_field = UserField()
+
+    def test_valid_username_accepted(self):
+        form_user = self.user_field.clean("admin")
+        self.assertEqual(form_user.email, self.user.email)
+        self.assertEqual(form_user.username, self.user.username)
+
+    def test_empty_username_when_required_throws_error(self):
+        with self.assertRaisesRegex(ValidationError, 'A user name or user ID is required.'):
+            self.user_field.clean('')
+
+    def test_none_username_when_required_throws_error(self):
+        with self.assertRaisesRegex(ValidationError, 'A user name or user ID is required.'):
+            self.user_field.clean(None)
+
+    def test_none_username_accepted_when_not_required(self):
+        self.user_field.required = False
+        self.assertIsNone(self.user_field.clean(None))
+        self.user_field.required = True
```

Comment:
Move the `.required = True` part in a `setUp` method and it will become the default for every test scenario. That would take effect even if the current test method fails and won't break subsequent methods.

Reply:
Ahh ok, thanks for this clever hint.